br/pkg/restore/snap_client/systable_restore.go

```diff
@@ -133,6 +147,145 @@ var unRecoverableSchema = map[string]struct{}{
 	mysql.WorkloadSchema: {},
 }
 
+type SchemaVersionPairT struct {
+	UpstreamVersionMajor   int64
+	UpstreamVersionMinor   int64
+	DownstreamVersionMajor int64
+	DownstreamVersionMinor int64
+}
+
+type schemaUpdateSQL struct {
+	schemaName string
+	tableName  string
+	sql        string
+}
+
+type versionSchemaUpdateSQL struct {
+	versionMajor int64
+	versionMinor int64
+	updateSQLs   []schemaUpdateSQL
+}
+
+var upgradeStatsTableSchemaList = []versionSchemaUpdateSQL{
```

Comment:
Does it mean every time after stats upgrade, we need add a upgrade sql here? how to make sure there's no lost sql for every upgrade?

Reply:
use `schemaversion` unit test to handle the change.